### PR TITLE
Support to run ImageMagick

### PR DESCRIPTION
**Pull request by csegarragonz**
In this PR I include support to run ImageMagick in WAVM, and WAMR. The SGX PR will follow, as it is larger in size.

I also add a test that takes a sample image and flips it. I use Faasm's logo as test image.

### CODE REVIEW

File: include/wasm/WasmEnvironment.h [hunk 1]
```diff
@@ -4,6 +4,20 @@
 #include <unordered_map>
 #include <vector>
 
+#define FAKE_NAME "faasm"
```
Comment (csegarragonz) [inline include/wasm/WasmEnvironment.h:7]: Moving these definitions to a header where all runtimes can benefit from them.

File: src/wamr/filesystem.cpp [hunk 2]
```diff
@@ -32,19 +33,47 @@ static uint32_t __wasi_fd_dup_wrapper(wasm_exec_env_t exec_env,
...
 static int32_t tempnam_wrapper(wasm_exec_env_t exec_env, int32_t a, int32_t b)
 {
     throw std::runtime_error("tempnam not implemented");
 }
 
 static NativeSymbol ns[] = {
-    REG_NATIVE_FUNC(__wasi_fd_dup, "(i*)i"),
-    REG_NATIVE_FUNC(dup, "(i)i"),
-    REG_NATIVE_FUNC(tempnam, "(ii)i"),
+    REG_NATIVE_FUNC(__wasi_fd_dup, "(i*)i"), REG_NATIVE_FUNC(dup, "(i)i"),
```
Comment (csegarragonz) [inline src/wamr/filesystem.cpp:74]: Weird formatting by `clang-format` here.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -64,6 +64,12 @@ jobs:
       - name: "Build libfake"
         run: ./bin/inv_wrapper.sh libfake
         working-directory: ${{ github.workspace }}/clients/cpp
+      - name: "Update ImageMagick submodule"
+        run: git submodule update --init -f third-party/ImageMagick
+        working-directory: ${{ github.workspace }}/clients/cpp
```
Comment (Shillaker) [inline .github/workflows/tests.yml:69]: Why do we need to update the ImageMagick submodule here? We're checking out the code with `submodules: true` which should do this implicitly.

**Comment by csegarragonz** [comment]
This should be good after a rebase with #688

**Pull request merged**